src/admin/api.js

```diff
@@ -212,11 +212,105 @@ async function setParticipantIsActiveFlag(endpoint, dfspName, value, logger) {
     return await put(buildUrl('participants', dfspName), payload, { endpoint, logger });
 }
 
+/**
+ * Communicates with the FXP API in order to fetch the FXP rates for all available currency channels .
+ *
+ * @method getFxpRatesPerCurrencyChannel
+ * @param {string} endpoint
+ * @param {object} logger
+ * @returns {object} An array of objects listing the exchange rates per currency channel, with this format:
+ *  {
+ *      <{string} the currency channel identifier. Format: concatenation of source and destination currencies, Example: "eurusd">: [
+ *          {
+ *              rate: <{int} The foreign exchange rate configured. The number of decimal points is provided in decimalRate.>,
+ *              decimalRate: <{int} The number of decimal points of the rate>,
+ *              startTime: <{string} The starting period during which the exchange rate can be applied to Forex transfers. Format: yyyy-MM-ddTHH:mm:ss.SSS[-HH:MM]>,
+ *              endTime: <{string} The ending period up to which the exchange rate can be applied to Forex transfers. Format: yyyy-MM-ddTHH:mm:ss.SSS[-HH:MM]>,
+ *              reuse: <{boolean} Whether to override the end date or not. When set to true and there is no other exchange rate valid at that point in time, it is expected that the exchange rate remains valid after the end date>
+ *          }
+ *      ]
+ *  }
+ */
+async function getFxpRatesPerCurrencyChannel(endpoint, logger) {
+    const currencyChannels = await getFxpCurrencyChannels(endpoint, logger);
+    const ratesForAllCurrencyChannels = await Promise.all(currencyChannels.map(async (currencyChannel) => {
+        return await getFxpRatesForChannel(endpoint, currencyChannel, logger);
+    }));
+    const ratesPerCurrencyChannel = {};
+
+    ratesForAllCurrencyChannels.forEach((item) => {
+        if (item.rates != null) {
```

Comment:
Interesting that the linter didn't catch this.
```suggestion
        if (item.rates !== null) {
```

Reply:
That is not an issue that the linter should catch, the `null` covers here all of "empty" cases and is not meant to be strictly equal to itself.

Reply:
Then use `!item.rates` instead.

Reply:
It is not the same.
```true != null //true
false != null //true
```
while
```
!false //true
!true //false```